hat/core/src/main/java/hat/ComputeContext.java

```diff
@@ -126,13 +126,30 @@ protected ComputeContext(Accelerator accelerator, Method computeMethod) {
      */
 
     public void dispatchKernel(int range, QuotableKernelContextConsumer quotableKernelContextConsumer) {
+        dispatchKernel(range, 0, 0, 1, quotableKernelContextConsumer);
+    }
+
+    public void dispatchKernel(int rangeX, int rangeY, QuotableKernelContextConsumer quotableKernelContextConsumer) {
+        dispatchKernel(rangeX, rangeY, 0, 2, quotableKernelContextConsumer);
+    }
+
+    public void dispatchKernel(int rangeX, int rangeY, int rangeZ, QuotableKernelContextConsumer quotableKernelContextConsumer) {
+        dispatchKernel(rangeX, rangeY, rangeZ, 3, quotableKernelContextConsumer);
+    }
+
+    private void dispatchKernel(int rangeX, int rangeY, int rangeZ, int dimNumber, QuotableKernelContextConsumer quotableKernelContextConsumer) {
         Quoted quoted = Op.ofQuotable(quotableKernelContextConsumer).orElseThrow();
         LambdaOpWrapper lambdaOpWrapper = OpWrapper.wrap(computeCallGraph.computeContext.accelerator.lookup,(JavaOp.LambdaOp) quoted.op());
         MethodRef methodRef = lambdaOpWrapper.getQuotableTargetMethodRef();
         KernelCallGraph kernelCallGraph = computeCallGraph.kernelCallGraphMap.get(methodRef);
         try {
             Object[] args = lambdaOpWrapper.getQuotableCapturedValues(quoted, kernelCallGraph.entrypoint.method);
-            NDRange ndRange = accelerator.range(range);
+            NDRange ndRange = null;
+            switch (dimNumber) {
+                case 1 -> ndRange = accelerator.range(rangeX);
+                case 2 -> ndRange = accelerator.range(rangeX, rangeY);
+                case 3 -> ndRange = accelerator.range(rangeX, rangeY, rangeZ);
+            }
```

Comment:
Should we add a default branch that throws an exception instead of initializing `ndRange` to null?

Reply:
I thought about this. However, this method is method is private, and only the public dispatch method sets the value. I will add the exception though.

Reply:
BTW, in near future we should have our own exceptions (e.g., `HATRuntimeException`). For this PR I will use the generic `RuntimeException`